### ISSUE TITLE
Add a `SpecialGalileanGroup`

### ISSUE THREAD

**Issue by kellertuer** [JuliaManifolds/LieGroups.jl#11]
When we have a tutorial on how to start with and implement e Lie group it might be nice to also transfer the `SpecialGalileanGroup` from 
https://github.com/JuliaRobotics/RoME.jl/blob/master/src/factors/Inertial/IMUDeltaFactor.jl
see also https://en.wikipedia.org/wiki/Galilean_transformation#Galilean_group

**Comment by Affie** [comment]
@kellertuer, I started looking at this but need help on the first step of defining the Lie Group.
 The Group is defined as `(SO(n) ⋉ ℝⁿ) ⋉ (ℝⁿ × ℝ)` so I tried the following and it seems to work except for a `show` error:
```julia
const LeftSpecialGalileanGroupOperation = LeftSemidirectProductGroupOperation{
    LeftSemidirectProductGroupOperation{
        MatrixMultiplicationGroupOperation,
        AdditionGroupOperation,
        LeftGroupOperationAction
    },
    ProductGroupOperation{
        Tuple{AdditionGroupOperation, AdditionGroupOperation}
    },
    LeftGroupOperationAction
}

const SpecialGalileanGroup{T} = LieGroup{
    ℝ,
    <:LeftSpecialGalileanGroupOperation,
    <:ProductManifold{
        ℝ,
        Tuple{
            <:Rotations{T},
            <:Euclidean{T,ℝ},
            <:Euclidean{T,ℝ},
            <:Euclidean{ManifoldsBase.TypeParameter{Tuple{1}},ℝ}}
        },
}

ProductTranslationTimeGroup{T} = LieGroup{ℝ, ProductGroupOperation{Tuple{AdditionGroupOperation, AdditionGroupOperation}}, ProductManifold{ℝ, Tuple{Euclidean{T, ℝ}, Euclidean{ManifoldsBase.TypeParameter{Tuple{1}}, ℝ}}}}
default_left_action(::SpecialEuclideanGroup, ::ProductTranslationTimeGroup) = LeftGroupOperationAction()
SpecialGalileanGroup(n::Int) = (SpecialOrthogonalGroup(n) ⋉  TranslationGroup(n)) ⋉ (TranslationGroup(n) × TranslationGroup(1))
```

Is this the correct way and is the `default_left_action` part correct?
Since I'm just using product and semidirect products of existing groups, I assumed that the above would be enough to implement for basic functions such as `inv` and `identity_element` if it was done correctly.

**Comment by kellertuer** [comment]
Yes ± my jet lag that does look ok. The const makes it nicer to dispatch and might need a few `<:` in the parameters.

And sure then you implementent all functions you need. I hope the API also mentions when functions have a default they use :)

Oh one thing I do not understand: If the action if the group operation of the left – you might still have to implement that here because SE(n) (and also only in homogeneous coordinates) would act on a vector from R^(n+1) not on a pair of vector of length n and 1.

**Comment by Affie** [comment]
Regarding the `LeftGroupOperationAction`: I was a bit confused as to what this refers to.
This is the current `SpecialEuclideanGroup`
```julia
julia> G = SpecialEuclideanGroup(2)
SpecialEuclideanGroup(2)

julia> G.op.action_type
LeftGroupOperationAction()
```
From docs https://juliamanifolds.github.io/LieGroups.jl/stable/interface/actions/#LieGroups.LeftGroupOperationAction 
> _"when acting on the group itself from the left"_

I might be misunderstanding it now or it may be a bug.
Is the `action_type` not the action of SO(n) on Tr(n), therefore it would have to be the rotation action from [Manifolds.jl](https://juliamanifolds.github.io/Manifolds.jl/stable/features/group_actions/#Manifolds.RotationAction) ( I couldn't find the equivalent in LieGroups.jl yet.)

Am I correct that the action I want would have to be something similar to one of these: https://juliamanifolds.github.io/Manifolds.jl/stable/features/group_actions/#Rotation-translation-action-(special-Euclidean)

**Comment by kellertuer** [comment]
One challenge with the current state of `GroupManifolds` (in `Manifolds.jl`) is, that there is a bit too many left/right “things” and some are also called (to me randomly) forward/backward, mainly to distinguish two left/right thingies.

Left group operation action refers to the following: Given some $g \in G$ then the function $a_g(p) = g\circ p$ can be defined for all $p$ where the group operation $\circ$ still makes sense. For the case of Matrix(-vector) multiplication, $g$ could be a matrix and $p$ still a vector and that would be fine. That is the case for `G.op.action_type`. And it is left, since one could also do some $b_g(p) = pg$ (whenever that is well-defined for some $p$), this would be a right action.

So maybe `LeftGroupOperationAction` has a bit of a Leeway here, since the elements we can “put” into $a_g$ are more than just other group elements $h$, but the operation (in sense of array multiplication) is still the same.

If you feel we should be more precise here, we can surely take over more action names. The code works with the current state as well.
Or in short both the `RotationAction` (which would here be called `LeftRotationAction` since we discontinued the direction parameter and just use names) as well as a `LeftRotationTranslationAction` (in homogeneous coordinates probably applied to a vector from $R^{n+1}$?) could be introduced for more clarity.

Also note that this `LeftGroupOperationAction` is also used for the `variant=:right` case of SE(n), which means that the variant really specifies the order of the pair (t,R) for right vs (R,t) for left) but the action used is always the left one so in the formulae only the positions of the elements change, not the actual formulae.

_edit:_ see for example https://juliamanifolds.github.io/Manifolds.jl/stable/features/group_actions/#Manifolds.apply_diff_group-Tuple{GroupOperationAction,%20Any,%20Any,%20Any} that we formerly had left/right actions as how they “behave” _and_ which side they came from. For now in this package we just fix the side they come from, not (yet?) how they “behave”. Compared to that doctoring the new Left here is the old left-left; the new right here is the old right-right ;)
